Yafc/Windows/MainScreen.PageListSearch.cs

```diff
@@ -0,0 +1,133 @@
+        private SearchNameMode searchNameMode = SearchNameMode.Both;
+
+        // Initialize both the current and previous states to searching by page name only.
+        public PageListSearch() => checkboxValues[0] = previousCheckboxValues[0] = true;
+
+        /// <summary>
+        /// Draws the search header for the page list dropdown.
+        /// </summary>
+        /// <param name="updatePageList">The action to perform if the user updates any of the search parameters.</param>
+        public void Build(ImGui gui, Action updatePageList) {
+            using (gui.EnterGroup(new Padding(1f))) {
+                if (gui.BuildSearchBox(query, out query)) {
+                    updatePageList();
+                }
+                gui.SetTextInputFocus(gui.lastContentRect, query.query);
+
+                gui.BuildText("Search in:");
+                using (gui.EnterRow()) {
+                    buildCheckbox(gui, "Page name", ref checkboxValues[(int)PageSearchOption.PageName]);
+                    buildCheckbox(gui, "Desired products", ref checkboxValues[(int)PageSearchOption.DesiredProducts]);
+                    buildCheckbox(gui, "Recipes", ref checkboxValues[(int)PageSearchOption.Recipes]);
+                }
+                using (gui.EnterRow()) {
+                    buildCheckbox(gui, "Ingredients", ref checkboxValues[(int)PageSearchOption.Ingredients]);
+                    buildCheckbox(gui, "Extra products", ref checkboxValues[(int)PageSearchOption.ExtraProducts]);
+                    if (gui.BuildCheckBox("All", checkboxValues.All(x => x), out bool checkAll)) {
+                        if (checkAll) {
+                            // Save the previous state, so we can restore it if necessary.
+                            Array.Copy(checkboxValues, previousCheckboxValues, (int)PageSearchOption.MustBeLastValue);
+                            Array.Fill(checkboxValues, true);
+                        }
+                        else {
+                            // Restore the previous state.
+                            Array.Copy(previousCheckboxValues, checkboxValues, (int)PageSearchOption.MustBeLastValue);
+                        }
+                        updatePageList();
+                    }
+                }
+                using (gui.EnterRow()) {
+                    buildRadioButton(gui, "Localized names", SearchNameMode.Localized);
+                    buildRadioButton(gui, "Internal names", SearchNameMode.Internal);
+                    buildRadioButton(gui, "Both", SearchNameMode.Both);
+                }
+            }
+
+            void buildCheckbox(ImGui gui, string text, ref bool isChecked) {
+                if (gui.BuildCheckBox(text, isChecked, out isChecked)) {
+                    if (checkboxValues.Any(x => x)) {
+                        updatePageList();
+                    }
+                    else {
+                        // Don't let the user uncheck the last checkbox.
```

Comment:
What I'm about to suggest is not mandatory for this PR to get merged, but it would improve the ease of use.

I suggest to allow having no checkboxes, and to do no search in this case. I suggest that because my first action was to to `disable one thing, enable another`, not the reverse. I believe many other players have the same habit.

Please let me know what you think about it.